Sources/JavaScriptKit/BasicObjects/JSDate.swift

```diff
@@ -0,0 +1,203 @@
+public final class JSDate {
+    private static let constructor = JSObject.global.Date.function!
+    private let ref: JSObject
+
+    public init(millisecondsSinceEpoch: Int? = nil) {
+        if let milliseconds = millisecondsSinceEpoch {
+            ref = Self.constructor.new(milliseconds)
+        } else {
+            ref = Self.constructor.new()
+        }
+    }
+
+    /** According to the standard, `monthIndex` is zero-indexed, where `11` is December. `day` 
+    represents a day of the month starting at `1`.
+    */
+    public init(
+        year: Int,
+        monthIndex: Int,
+        day: Int = 1,
+        hours: Int = 0,
+        minutes: Int = 0,
+        seconds: Int = 0,
+        milliseconds: Int = 0
+    ) {
+        ref = Self.constructor.new(year, monthIndex, day, hours, minutes, seconds, milliseconds)
+    }
+
+    /// Year of this date in local time zone.
+    public var fullYear: Int {
+        get {
+            Int(ref.getFullYear.function!().number!)
```

Comment:
I think this can just be `Int(ref.getFullYear!().number!)`. Same for the other function calls